src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -16281,6 +16281,108 @@ static void Main()
         }
 
         [Fact]
+        public void IsNull()
+        {
+            var source =
+@"class C
+{
+    static void F1(object o) { }
+    static void F2(object o) { }
+    static void G(object? o)
+    {
+        if (o is null)
+        {
+            F1(o);
+        }
+        else
+        {
+            F2(o);
+        }
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (9,16): warning CS8604: Possible null reference argument for parameter 'o' in 'void C.F1(object o)'.
+                //             F1(o);
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "o").WithArguments("o", "void C.F1(object o)").WithLocation(9, 16));
+        }
+
+        [Fact]
+        public void IsInvalidConstant()
+        {
+            var source =
+@"class C
+{
+    static void F(object o) { }
+    static void G(object? o)
+    {
+        if (o is F)
+        {
+            F(o);
+        }
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (6,18): error CS0428: Cannot convert method group 'F' to non-delegate type 'object'. Did you intend to invoke the method?
+                //         if (o is F)
+                Diagnostic(ErrorCode.ERR_MethGrpToNonDel, "F").WithArguments("F", "object").WithLocation(6, 18),
+                // (8,15): warning CS8604: Possible null reference argument for parameter 'o' in 'void C.F(object o)'.
+                //             F(o);
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "o").WithArguments("o", "void C.F(object o)").WithLocation(8, 15));
+        }
+
+        [Fact]
+        public void IsPattern()
+        {
+            var source =
+@"class C
+{
+    static void F(string s) { }
+    static void G(object? o)
+    {
+        if (o is string s)
+        {
+            F(s);
+        }
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void PatternSwitch()
+        {
+            var source =
+@"class C
+{
+    static void F(object o) { }
+    static void G(object? x)
+    {
+        switch (x)
+        {
+            case string s:
+                F(s);
+                break;
+            case object y when y is string t:
+                F(y);
+                F(t);
+                break;
+            case null:
+                F(x);
```

Comment:
>F(x); [](start = 16, length = 5)

It would be interesting to test what happens when the same call is used in other switch sections.

Reply:
Good catch, thanks. In those cases, `x` is treated (incorrectly) as nullable. I've updated the test and will fix in subsequent PR.